app/pages/notifications/notification-section.jsx

```diff
@@ -143,13 +144,13 @@ export default class NotificationSection extends Component {
     const { notificationsCounter } = this.context;
     const { user } = this.props;
     const { notifications } = this.state;
-    const relatedNotifications = notifications.filter(
-      notification => notification.source_id === readNotification.source_id
-    );
-    relatedNotifications.push(readNotification);
-    const readRequests = relatedNotifications
+    const relatedNotifications = notifications.filter(notification => (
+      notification.subscription_id === readNotification.subscription_id
+        && notification.delivered === false
+    ));
+    const promises = relatedNotifications
```

Comment:
`promises` doesn't tell us what this variable is doing. Is there a better name that could be used here?